fire/testutils.py

```diff
@@ -34,14 +34,41 @@ class BaseTestCase(unittest.TestCase):
   """Shared test case for Python Fire tests."""
 
   @contextlib.contextmanager
-  def assertStdoutMatches(self, regexp):
-    """Asserts that the context generates stdout matching regexp."""
-    stdout = six.StringIO()
-    with mock.patch.object(sys, 'stdout', stdout):
-      yield
-    value = stdout.getvalue()
-    if not re.search(regexp, value, re.DOTALL | re.MULTILINE):
-      raise AssertionError('Expected %r to match %r' % (value, regexp))
+  def assertOutputMatches(self, stdout='.*', stderr='.*', capture=True):
+    """Asserts that the context generates stdout and stderr matching regexps.
+
+    Args:
+      stdout (str): regexp to match against stdout (None will check no stdout)
```

Comment:
I was following this guide :) - http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html